### PR TITLE
Add Apple visionOS support

### PR DESCRIPTION
**Pull request by QuentinPerez**
## Motivation

Add the support of Apple VisionOS  [121419](https://github.com/rust-lang/rust/pull/121419)

## Solution
Add target_os = "visionos"

### CODE REVIEW

**Comment by Thomasdezeeuw** [comment]
Can you add the target to the CI and local dev?
* https://github.com/rust-lang/socket2/blob/c93cdcc25fd3de7ef20515b9c16ed2b5771ed268/.github/workflows/main.yml#L59
* https://github.com/rust-lang/socket2/blob/c93cdcc25fd3de7ef20515b9c16ed2b5771ed268/Makefile#L3

**Comment by QuentinPerez** [comment]
@Thomasdezeeuw do you mind if I use `target_env = "apple"` ? Instead of changing this every time Apple releases a new device

**Comment by Thomasdezeeuw** [comment]
> @Thomasdezeeuw do you mind if I use `target_env = "apple"` ? Instead of changing this every time Apple releases a new device

I think that is deprecated. Or was that `target_family`?

**Comment by Thomasdezeeuw** [comment]
> > @Thomasdezeeuw do you mind if I use `target_env = "apple"` ? Instead of changing this every time Apple releases a new device
> 
> I think that is deprecated. Or was that `target_family`?

`target_vendor` is being deprecated: https://github.com/rust-lang/rust/issues/100343. For now let's stick with explicit OS, I don't really want to deal with all the target stuff a the moment, at least we know that this works.

**Comment by madsmtm** [comment]
`target_vendor = "apple"` will not be deprecated unless an alternative appears, see https://github.com/rust-lang/rust/issues/100343#issuecomment-2065159934, so I think you can safely use that

**Comment by Thomasdezeeuw** [comment]
> `target_vendor = "apple"` will not be deprecated unless an alternative appears, see [rust-lang/rust#100343 (comment)](https://github.com/rust-lang/rust/issues/100343#issuecomment-2065159934), so I think you can safely use that

Let's still with `target_os` for now, we'll let the dust settle on vendor vs. family vs. os, etc. first.

File: .cirrus.yml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 freebsd_instance:
-  image_family: freebsd-13-2
+  image_family: freebsd-13-3
```
Comment (QuentinPerez) [inline .cirrus.yml:2]: I changed it due to the not found error on freebsd-13-2

**Comment by QuentinPerez** [comment]
@Thomasdezeeuw do you think that we can merge ?

**Comment by Thomasdezeeuw** [comment]
I want to merge #520 first, then it's a lot easier to add new targets.

**Comment by Darksonn** [comment]
#520 has been merged now.

**Review by Thomasdezeeuw — APPROVED** [review]

**Comment by Thomasdezeeuw** [comment]
Thanks @QuentinPerez

**Pull request merged**